python/ray/serve/tests/unit/test_deployment_state.py

```diff
@@ -2862,6 +2868,109 @@ def test_deploy_with_partial_constructor_failure(mock_deployment_state_manager):
     )
 
 
+def test_deploy_with_placement_group_failure(mock_deployment_state_manager):
+    """
+    Test deploy with a placement group failure.
+    """
+
+    create_dsm, _, _ = mock_deployment_state_manager
+    dsm: DeploymentStateManager = create_dsm()
+
+    def fake_create_placement_group_fn(placement_group_bundles, *args, **kwargs):
+        """Fakes the placement_group_fn used by the scheduler.
+
+        Lets the test to run without starting Ray. Raises an exception if the
+        bundles are invalid.
+        """
+
+        validate_placement_group(bundles=placement_group_bundles)
+
+    dsm._deployment_scheduler._create_placement_group_fn = (
+        fake_create_placement_group_fn
+    )
```

Comment:
Can we pass the fake function in when constructing the deployment state manager?

Reply:
I made the change. Note that `DeploymentStateManager` doesn't currently have a `create_placement_group_fn_override` argument. I added one, and then passed in the fake in the test.